lib/sgpres.gi

```diff
@@ -2766,38 +2766,1163 @@ end);
 ##  augmented coset table data. It must not be applied to augmented coset
 ##  tables which are not of type 2.)
 InstallGlobalFunction(CopiedAugmentedCosetTable,function(aug)
-local t;
-  t:=rec(
-	  isAugmentedCosetTable:=true,
-	  type:=aug.type,
-	  tableType:=aug.tableType,
-	  groupGenerators:=aug.groupGenerators,
-	  groupRelators:=aug.groupRelators,
-	  cosetTable:=aug.cosetTable,
-	  cosetFactorTable:=aug.cosetFactorTable,
-	  primaryGeneratorWords:=aug.primaryGeneratorWords,
-	  tree:=aug.tree,
-	  treeNumbers:=aug.treeNumbers,
-	  numberOfSubgroupGenerators:=aug.numberOfSubgroupGenerators,
-	  nameOfSubgroupGenerators:=aug.nameOfSubgroupGenerators,
-	  subgroupGenerators:=aug.subgroupGenerators
-	);
-  if IsBound(aug.conversionList) then
-    t.conversionList:=aug.conversionList;
+local t,j;
+  if IsBound(aug.isNewAugmentedTable) then
+    t:=rec(isNewAugmentedTable:=true);
+    for j in 
+      [ "A", "aug", "ct", "defcount", "from", "homgenims", "homgens",
+      "index", "n", "offset", "primaryImages", "rels",
+      "secondary", "secount", "secondaryImages", "subgens" ] do
+      if IsBound(aug.(j)) then
+	t.(j):=aug.(j);
+      fi;
+    od;
+  else
+    # old version
+    t:=rec(
+	    isAugmentedCosetTable:=true,
+	    type:=aug.type,
+	    tableType:=aug.tableType,
+	    groupGenerators:=aug.groupGenerators,
+	    groupRelators:=aug.groupRelators,
+	    cosetTable:=aug.cosetTable,
+	    cosetFactorTable:=aug.cosetFactorTable,
+	    primaryGeneratorWords:=aug.primaryGeneratorWords,
+	    tree:=aug.tree,
+	    treeNumbers:=aug.treeNumbers,
+	    numberOfSubgroupGenerators:=aug.numberOfSubgroupGenerators,
+	    nameOfSubgroupGenerators:=aug.nameOfSubgroupGenerators,
+	    subgroupGenerators:=aug.subgroupGenerators
+	  );
+    if IsBound(aug.secondaryWords) then
+      t.secondaryWords:=Immutable(aug.secondaryWords);
+    fi;
+
+    if IsBound(aug.conversionList) then
+      t.conversionList:=aug.conversionList;
+    fi;
+    if IsBound(aug.primarySubgroupGenerators) then
+      t.primarySubgroupGenerators:=Immutable(aug.primarySubgroupGenerators);
+    fi;
+    if IsBound(aug.subgroupRelators) then
+      t.subgroupRelators:=Immutable(aug.subgroupRelators);
+    fi;
+    if IsBound(aug.translationTable) then
+      t.translationTable:=Immutable(aug.translationTable);
+    fi;
+
+  fi;
+  return t;
+end);
+
+
+# New version of MTC based on DFH's book
```

Comment:
This is one of the very, very few comments for the old code. Could we please spell out the full title and author of the book in question? That way, somebody who needs to edit the MTC code in a few years at least has a fighting chance :-).